### DISCUSSION TITLE
Trilium managed or hosted wiki?

### DISCUSSION THREAD

**Discussion by maphew** [TriliumNext/Notes#62] [Ideas]
> ...we should probably add an issue for either replicating the wiki or providing a docs site for the same stuff.
_from #16._

How much work to migrate [zadam/wiki ](https://github.com/zadam/trilium/wiki) to a Trilium hosted or managed wiki?

Two paths I can see off the top of my head:
- straight up hosting, using the builtin [Sharing](https://github.com/zadam/trilium/wiki/Sharing) feature. Would require a server somewhere and someone to pay for compute time. There might be a path via Clouldflare Hosting, Fly.io, ... that offer a free tier.
- leverage Trilium External API and export to markdown/html and publish via Github Actions to GH Pages
- clone _zadam/wiki_ to this org and carry on using Github Wiki tools

Are there others?

**Comment by maphew** [comment]
Outside of how the content is exposed to general public, the 1st major challenge for a Trilium backed wiki is lack of multi-user or shared note branches support. Maybe the desire to share wiki could be a or the driver to making that happen.

An interim measure could a shared secret in this repo with the trilium server password, and people access the trilium server using browser instead of desktop or mobile app. Lots of holes in this idea, it could fall apart quickly, but, I think it could work, for a time.

**Comment by meichthys** [comment]
In the short term it would probably be easiest to clone the existing Trilium Wiki, but once the building blocks are in place we should definitely aim to use TriliumNext to host its own documentation.

**Reply by zerebos** [reply]
It's definitely easy enough to do this considering [trilium.rocks](https://trilium.rocks) is running through Trilium's share feature on a low-power server with cloudflare.

**Reply by maphew** [reply]
is that running on [cloudlare hosting](https://www.cloudflare.com/en-ca/developer-platform/hosting/) or cloudflare caching in front the server that's hosting it? I'd very interested in details on the former

**Comment by maphew** [comment]
grump. Github support for forking and merging of wiki doesn't exist. There are a few recipes out there, but none are nice ([1](https://gist.github.com/subfuzion/0d3f19c4f780a7d75ba2), [2](https://gist.github.com/larrybotha/10650410), [3](https://stackoverflow.com/questions/7197937/on-github-can-i-fork-just-a-wiki/56480628#56480628)). IMO the least amount of messing about is: 

* Github >> [new repo](https://github.com/new) >> Import repository, and use `https://github.com/zadam/trilium.wiki.git` as source.

This will give a clean source to use for _Trilium >> Import_ function later, and PR's can be accepted from wherever starting immediately. I don't think creating a local wiki and importing the content gains very much.

Here's an example: https://github.com/maphew/zadam-trilium-wiki, but it would be better if someone with _create repositories_ permission do this under the Trilium Next umbrella.

**Reply by maphew** [reply]
>...creating a local wiki 

I meant a wiki as part of the TriliumNext/Notes repo, not in the sense of a person's local computer.

**Comment by Alumniminium** [comment]
I wouldn't mind manually porting the Wiki until we have a wiki inside trilium next. I lack the permissions to create the wiki in the Notes repo.

**Reply by meichthys** [reply]
This sounds like a good start to me. I'm also wondering if it would be a good idea to port the wiki to regular html pages that we can host using github pages? It may be a bit more effort, but that would allow us to accept pull requests against it (currently there is no way to submit a pull request for a github wiki afaik).

**Reply by zerebos** [reply]
It wouldn't be a bad idea to have a `docs` repo with all the markdown. From there we could shape it into a nicer webpage on github pages with one of the doc builders like docusaurus, vitepress, mkdocs, gitbook, or similar.

**Reply by meichthys** [reply]
@rauenzi that sounds like a good idea. a markdown directory would also be very easy to import into a TriliumNext instance if we decide to host the wiki on TriliumNext at some point 

@Alumniminium Is this something you would be willing to do for us? (Convert the existing Trilium wiki into markdown files in a new TriliumNext 'docs' repo? I'd be glad to set the repo up and provide you access to it.

**Reply by zerebos** [reply]
> a markdown directory would also be very easy to import into a TriliumNext instance if we decide to host the wiki on TriliumNext at some point

My thoughts exactly! That means we can use whichever fancy doc builder we want as long as we keep the actual docs themselves in markdown.

**Reply by Alumniminium** [reply]
@meichthys yeah that's fine with me. Any special wishes for where to put the images? Where do you want the docs to be?

Here: https://github.com/TriliumNext/Notes/tree/develop/docs, in a wiki folder in the root or in a separate repo?

**Reply by meichthys** [reply]
I suppose a separate repo may be best since some of the wiki covers the usage of Trilium Web Clipper and the Mobile app.

**Reply by meichthys** [reply]
@Alumniminium I setup a `Docs` repo and have given you maintainer privs, so you should be good to go.
As for the images, I don't really have a preference. I would say whatever you think is best is fine. We can always re-structure if we need to down the road.

**Reply by meichthys** [reply]
I've created an issue in the Docs repo to track this:
https://github.com/TriliumNext/Docs/issues/1
Feel free to use that issue to track your progress or address other questions/issues you may run into.

**Reply by maphew** [reply]
This is what a naive import as-is no changes of Zadam Wiki exported as a zip file looks like. Ignoring the navigation, it's serviceable:

![image](https://github.com/TriliumNext/Notes/assets/486200/36c6e0d4-2ba8-477f-b369-b8f1f751d837)

Once @Alumniminium has ported the files into Docs repo per https://github.com/TriliumNext/Docs/issues/1 it doesn't look like that much work to rearrange so the referenced media don't swarm the table of contents.

**Reply by meichthys** [reply]
Doesn't look to bad. We could probably also touch base with @rauenzi to see how he setup [TriluimRocks](https://trilium.rocks/). It looks like some nice share theming and no doubt some css wizardry.

**Reply by Alumniminium** [reply]
I will be doing that as soon as possible. The company I work for just went bankrupt and we're reorganizing into a new company right now, so I'm a bit preoccupied. I hope to start porting tonight and I plan to finish it over the weekend at the latest.

**Reply by Alumniminium** [reply]
![240605_21h21m13s_screenshot](https://github.com/TriliumNext/Notes/assets/23240553/3023faa5-49fa-40b7-952e-fc7129226651)
![240605_21h23m04s_screenshot](https://github.com/TriliumNext/Notes/assets/23240553/f4fe8058-7f63-44aa-b1ce-5c6a34ad9de7)

@maphew I've fixed the links and images in [Docs](https://github.com/TriliumNext/Docs). You should be able to import it in trilium. I won't have time to do more until the weekend, so feel free to jump in and push to the repo.

**Reply by maphew** [reply]
It's really encouraging to see this moving along! I'm offline tomorrow for sure and possibly next few days also. After that I'll definitely jump in.

**Reply by zerebos** [reply]
> Doesn't look to bad. We could probably also touch base with @rauenzi to see how he setup [TriluimRocks](https://trilium.rocks/). It looks like some nice share theming and no doubt some css wizardry.

Someone in Discord also asked about this recently, so I took the time to write up a guide on how to use the template I use for Trilium Rocks, but it also contains some technical details if someone wanted to make something similar but not the same.

Guide here: https://trilium.rocks/xecUJ8eL3pvM

**Reply by Alumniminium** [reply]
> Someone in Discord also asked about this recently, so I took the time to write up a guide on how to use the template I use for Trilium Rocks, but it also contains some technical details if someone wanted to make something similar but not the same.
> 
> Guide here: https://trilium.rocks/xecUJ8eL3pvM

might as well do that too, weekend gonna be busy :D

**Reply by zerebos** [reply]
I managed to finally import the docs (I'll make a PR to the repo soon to fix some things) and apply my template to it. Here's an example of how it may look without any customization (https://notes.zerebos.com/share/wiki):

![image](https://github.com/TriliumNext/Notes/assets/6865942/1fdc31a4-489d-4a53-a91e-1d409ebb05bc)

Also, I'll probably start integrating more and more info from the wiki into `trilium.rocks`, so that could be a potential option for the docs if we wanted to go that route.

**Reply by eliandoran** [reply]
@rauenzi , I would like to reproduce the wiki with the same share format as trilium.rocks (similar to the link you have provided), do you have any instructions on how to do that?

**Reply by zerebos** [reply]
I actually made a write-up on how to do so a while ago https://trilium.rocks/xecUJ8eL3pvM. The setup is a little cumbersome currently, but with the next version it will be much simpler (just importing a zip ideally).

Of course if you want any help or customizations let me know!

**Comment by meichthys** [comment]
https://github.com/TriliumNext/Docs/issues/2 is tracking the TOC/Readme.

**Comment by meichthys** [comment]
Closing this discussion since work is well underway in the [Docs repo](https://github.com/TriliumNext/Docs) and the [Initial Documentation Review Project](https://github.com/orgs/TriliumNext/projects/4). 

Also relevant: https://github.com/TriliumNext/Notes/issues/339
